cypress/integration/project_checkout.spec.js

```diff
@@ -0,0 +1,102 @@
+import { DIALOG_SHOWOPENDIALOG, OPEN_PATH } from "../../native/ipc.js";
+
+const withWorkspaceStub = callback => {
+  cy.exec("pwd").then(result => {
+    const pwd = result.stdout;
+    const checkoutPath = `${pwd}/cypress/workspace/checkout`;
+
+    // Clean up any left-overs from previous failed tests.
+    cy.exec(`rm -rf ${checkoutPath}`);
```

Comment:
Is there a way to specify setup and teardowns in general for cypress?

Scrolling down I see `beforeEach`. Should these commands live there?

Reply:
Possibly. I'm using the same pattern here as in https://github.com/radicle-dev/radicle-upstream/blob/master/cypress/integration/project_creation.spec.js.

At an earlier point I tried to extract this into a cypress command, but it didn't work as I imagined because of subtle differences on how Cypress is doing async: https://docs.cypress.io/guides/core-concepts/introduction-to-cypress.html#Commands-Are-Not-Promises.

Extracting this functionality into a wrapper function was the only way I could get it to work reliably. Not sure if sinking more time into making this prettier makes sense.

Reply:
> Not sure if sinking more time into making this prettier makes sense.

I wasn't thinking prettier, I was thinking more reliable. If commands that are needing during setup are grouped outside at least they're localised to one place and can be used if we needed other tests that would need to do the same.

But if it's a PITA, then now worries

Reply:
> I wasn't thinking prettier, I was thinking more reliable.

Sure. But yeah, let's do this another time.